bfdriver/src/platform/efi/efi_base/base.c

```diff
@@ -0,0 +1,115 @@
+/*
+ * Bareflank Hypervisor
+ * Copyright (C) 2018 Assured Information Security, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies
+ * of the Software, and to permit persons to whom the Software is furnished to do
+ * so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+#include "base.h"
+#include "boot.h"
+#include "mp_service.h"
+#include "bfelf_loader.h"
+#include "common.h"
+
+
+EFI_MP_SERVICES_PROTOCOL *g_mp_services;
+extern char target_vmm_start[];
+extern char target_vmm_end[];
+
+
+VOID __attribute__((ms_abi)) bf_start_hypervisor_on_core(VOID *data)
+{
+    (void)data;
+    _set_ne();
+    Print(L"Starting hypervisor on core.\n");
```

Comment:
We should create a wrapper for this. We currently use this:

https://github.com/Bareflank/hypervisor/blob/master/bfsdk/include/bfdebug.h#L1098

Also.... I see a lot of print statements in this code, which is really noisy. Which statements are requirements?